test/Api.Tests/ControllerTests/TopicsControllerAttachmentsTest.cs

```diff
@@ -0,0 +1,254 @@
+        /// </summary>
+        [Fact]
+        public void GetAttachmentsTest403()
+        {
+            _tester.TestControllerWithMockData("newuser@hipapp.de")
+                .Calling(c => c.GetAttachments(_tester.TopicOne.Id))
+                .ShouldReturn()
+                .StatusCode(403);
+        }
+
+        /// <summary>
+        /// Returns ok if the specified attachment of the topic is retrieved
+        /// </summary>
+        [Fact]
+        public void GetAttachmetTest()
+        {
+            _tester.TestControllerWithMockData()
+                .WithDbContext(dbContext => dbContext
+                    .WithSet<TopicAttachment>(db => db.Add(TopicAttachment)))
+                .Calling(c => c.GetAttachmet(_tester.TopicOne.Id, TopicAttachment.Id))
+                .ShouldReturn()
+                .Ok()
+                .WithModelOfType<StringWrapper>()
+                .Passing(actual => actual.Value = "D704A706C781811D7143B2D1C980DCCD");
+        }
+
+        /// <summary>
+        /// Returns 404 if the specified attachment of the topic is not found
+        /// </summary>
+        [Fact]
+        public void GetAttachmetTest404()
+        {
+            _tester.TestControllerWithMockData()
+                .Calling(c => c.GetAttachmet(_tester.TopicOne.Id, TopicAttachment.Id))
+                .ShouldReturn()
+                .NotFound();
+        }
+
+        /// <summary>
+        /// Returns 403 if attachments are retrieved by other user
+        /// </summary>
+        [Fact]
+        public void GetAttachmetTest403()
+        {
+            _tester.TestControllerWithMockData("newuser@hipapp.de")
+                .Calling(c => c.GetAttachmet(_tester.TopicOne.Id,TopicAttachment.Id))
+                .ShouldReturn()
+                .StatusCode(403);
+        }
+
+        #endregion
+
+        #region post attachments
+        //Positive test case not possible as we need to attach the file
+
+        /// <summary>
+        /// Returns ok if an attachment is created for the topic
+        /// </summary>
+        [Fact]
+        public void PostAttachmentTest()
+        {
+            _tester.TestControllerWithMockData()
+                .Calling(c => c.PostAttachment(_tester.TopicOne.Id, AttachmentFormModel))
+                .ShouldHave()
+                .DbContext(db => db.WithSet<TopicAttachment>(relations =>
+                    relations.Any(actual => actual.Title == AttachmentFormModel.Title)))
+                .AndAlso()
+                .ShouldReturn()
+                .Ok();
+        }
+
+        /// <summary>
+        /// Returns unknown error if some problem occurs while adding an attachment
+        /// </summary>
+        [Fact]
+        public void PostAttachmentTest500()
+        {
+            _tester.TestController()
+                .Calling(c => c.PostAttachment(_tester.TopicOne.Id, AttachmentFormModel))
+                .ShouldReturn()
+                .StatusCode(500);
```

Comment:
Also, the test description does not really explain what actually happens and why